index.ts

```diff
@@ -374,13 +378,16 @@ function traverse<T extends MapResultKey>(
         const nodeSkip = verifySkip(name, skip);
 
         if (nodeSkip !== true) {
-            root[name] = root[name] || (nodes.length ? {} : false);
-            nodes.length && traverse(
-                nodes,
-                root[name],
-                opts,
-                nodeSkip,
-            );
+            const value = root[name] = root[name] || (nodes.length ? {} : false);
```

Comment:
What is the reason for this change?

Reply:
Typescript does not see the connection between `nodes.length` and `root[name]` type. 
However, if the assignment is extracted to its own variable, typescript can then tell (via if), that `value` is `MapResult` and not `false`, thus make the assignment valid.

It could be also rewritten to: (maybe nicer)
```ts
if (nodes.length) {
  traverse(
    nodes,
    root[name] = {},
    opts,
    nodeSkip,
  );
} else {
   root[name] = false;
}
```

Reply:
I would actually root for the new proposed code as it is more readable.